Add overflow check to encoding.DecodeVarint.

sbunce opened the pull request:
After we use DecodeUvarint to decode a positive signed int we need to overflow
check to make sure the resulting uint64 doesn't overflow int64.

sbunce committed:
Add overflow check to encoding.DecodeVarint.

After we use DecodeUvarint to decode a positive signed int we need to overflow
check to make sure the resulting uint64 doesn't overflow int64.

sbunce committed:
Add overflow check to encoding.DecodeVarint.

After we use DecodeUvarint to decode a positive signed int we need to overflow
check to make sure the resulting uint64 doesn't overflow int64.

petermattis commented on util/encoding/encoding_test.go:
```diff
@@ -362,6 +362,24 @@ func TestDecodeInvalidUvarint(t *testing.T) {
+			defer func() {
+				if r := recover(); r == nil {
+					t.Error("expected panic")
+				}
```
The tests which catch panics could be made mildly more robust by checking that the error message matches a regexp.

sbunce replied:
This is a good idea. I have the feeling that doing this in tests would encourage descriptive errors.

Done.

petermattis commented:
LGTM

sbunce committed:
Add regexps to check panic strings.

sbunce committed:
Add regexps to check panic strings.

petermattis commented:
Nice. LGTM.

petermattis commented:
@sbunce I've added you to the cockroach organization so you can merge this when ready.

tamird commented on util/encoding/encoding_test.go:
```diff
@@ -346,18 +348,71 @@ func TestEncodeDecodeUvarint(t *testing.T) {
-		// length of 9 bytes should cause an error.
-		{0x11, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01},
+	tests := []struct{
+		name    string // name printed with errors.
```
nit: this struct could be extracted. :+1: otherwise

petermattis replied:
You could also combine the tests by making the decode function a parameter of the struct:

```
  tests := []struct {
    ...
    decode func([]byte)
  }{
    {
      ...
      decode: func(b []byte) { DecodeUvarint(b) },
    },
  }
```

kkaneda committed:
A tiny comment fix: s/ReadUncertaintyIntervalError/ReadWithinUncertaintyIntervalError/

petermattis committed:
Don't rebuild everything when STATIC=1 is specified.

Only rebuild everything for the "release" target.

petermattis committed:
Merge pull request #957 from cockroachdb/pmattis/make-static

Don't rebuild everything when STATIC=1 is specified.

BramGruneir committed:
Adds the ability to request a single node or store status.

tbg committed:
use clog, our new fork of glog

BramGruneir committed:
Merge pull request #959 from cockroachdb/bram/status

Adds the ability to request a single node or store status.

tbg committed:
remove obsolete util/sampling.* code

tbg committed:
Merge pull request #963 from tschottdorf/clog

use clog, our fork of glog

kkaneda committed:
Merge pull request #948 from cockroachdb/kkaneda/comment_fix2

A tiny comment fix: s/ReadUncertaintyIntervalError/ReadWithinUncertaintyIntervalError/

bdarnell committed:
Merge pull request #964 from bdarnell/contrib

Update CONTRIBUTING.md prerequisites.

sbunce committed:
Add decode func data member and combine tests.

Combine panic tests for DecodeUvarint, DecodeVarint, DecodeBytes,
DecodeBytesDecreasing.

sbunce committed:
Merge branch 'master' of https://github.com/sbunce/cockroach

Conflicts:
	util/encoding/encoding_test.go

sbunce commented:
Sorry I'm still learning git. My git history for this is not right.

I'm going to close this one and start over.
